giraffe/src/utils/getTicks.ts

```diff
@@ -126,11 +126,16 @@ export const generateTicks = (
   tickStep: number
 ): number[] => {
   const generatedTicks = []
-  const [start, end] = domain
+  const [start = 0, end = 0] = domain
   const stepStart = isFiniteNumber(tickStart) ? tickStart : start
-  const step = isFiniteNumber(tickStep)
-    ? tickStep
-    : (end - stepStart) / (totalTicks + 1)
+
+  let step = tickStep
+  if (!isFiniteNumber(tickStep)) {
+    const parts =
+      isFiniteNumber(totalTicks) && totalTicks !== 0 ? totalTicks : 1
+    step = (end - stepStart) / (isFiniteNumber(tickStart) ? parts : parts + 1)
```

Comment:
```ts
isFiniteNumber(tickStart) ? parts : parts + 1
```

if `parts` isn't a finite number, what effect does adding 1 to it have? i mean the tests seem to indicate this is fine, but as a random dude reading this, that isn't super clear.

Reply:
The expression you quoted is checking `tickStart` for being a finite number.

To answer your question, let's briefly go over what tick marks do to a graph. Let's just look at the y-axis as an example. If we have just 1 tick mark on the entire y-axis, this tick mark effectively divides the graph into 2 "parts" - a top half and a bottom half. Having 2 tick marks divides the graph into thirds. 3 tick marks divides into quarters, etc.

So, to answer your question... if `tickStart` is finite, which means the user has specified a number for `tickStart`, we want to divide the graph (ie create a "step") by the number of `parts` which is already determined by this point in the code. If `tickStart` is not finite, which means the user has not specified it, then we want to divide the graph (create a "step") according to algorithm described in the paragraph above. Notice how the "parts" are always 1 more than the number of ticks. That's where the + 1 comes from.

Reply:
This may be difficult to visualize with just words. It's easier to demonstrate by running the code in storybook and using the "User defined ticks" story.